src/irmin-pack/unix/ext.ml

```diff
@@ -318,6 +310,31 @@ module Maker (Config : Conf.S) = struct
             | Some { use_auto_finalisation = true; _ } ->
                 let* _ = finalise_exn ~wait:false t in
                 Lwt.return_unit
+
+          let oldest_live_commit t =
+            let pl = Control.payload (File_manager.control t.fm) in
+            match pl.status with
+            | From_v1_v2_post_upgrade _ | Used_non_minimal_indexing_strategy
+            | No_gc_yet ->
+                None
+            | T1 | T2 | T3 | T4 | T5 | T6 | T7 | T8 | T9 | T10 | T11 | T12 | T13
+            | T14 | T15 ->
+                assert false
+            | Gced { oldest_live_commit_offset = offset; _ } -> (
+                let entry =
+                  Commit.CA.read_and_decode_entry_prefix ~off:offset
+                    t.dispatcher
+                in
+                match Commit.CA.Entry_prefix.total_entry_length entry with
+                | None ->
+                    (* Stores which support gc, also have a length in the header
+                       of all their entries. *)
+                    assert false
```

Comment:
The comment was untrue, I changed it, thanks.

Are we going to support GC on with V1 commits as target? I missed that. The code is not ready for this I think, so I propose we  change this when we do add the Gc support on V1.

> the irmin-pack functor config allows contents with no length stored. 

 I changed the comment to only mention commits (which are the target of this operation).

Reply:
> Are we going to support GC on with V1 commits as target

Nothing would prevent an irmin-pack user to launch a GC targeting a V1 commit. However, it will not occur in Tezos because V1 commits haven't been generated for months and tezos only targets commits ~15 days in the past.

However, we might not have to handle this for commit objects but we will have to handle it for inode objects. A recent block most likely still targets 1 y/o inodes.

> The code is not ready for this I think

I checked and it's partially handled. I think we should open an issue to not forget.

Reply:
> Nothing would prevent an irmin-pack user to launch a GC targeting a V1 commit. However, it will not occur in Tezos ..

I think we are in agreement, that we will not add support for this (i.e. calling gc on V1 commits)? (I was only referring to V1 commits in my previous comment, as this code only targets commits).

Reply:
Sorry if I'm being unclear. 

On master:
- The code supports V1 gc-targets but the CF forbids it thanks to the `from_v1_v2` tag
- The code doesn't support V1 object in the tree of the GC target.

After your PR:
- The code doesn't support V1 gc-targets, but that's OK since the CF already forbits this
- Same for V1 inodes/contents

When implementing the lower layer:
- We will have to choose if we support V1 gc-targets. Either we do or we raise a new dedicated error.
- We will have to support V1 inodes/contents in the tree of the GC target

Reply:
ok, thanks